From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Rose <japthorp@slack-corp.com>
Date: Thu, 29 Sep 2022 16:30:17 -0700
Subject: fix parallel/test-v8-stats

Refs https://chromium-review.googlesource.com/c/v8/v8/+/3967841
Refs https://chromium-review.googlesource.com/c/v8/v8/+/3892950.

Heap spaces were added/removed in v8 - this patch updates test expectations
to match.

This patch can be removed when Electron updates to Node.js v20.

diff --git a/test/parallel/test-v8-stats.js b/test/parallel/test-v8-stats.js
index 2eaa3c5b0609149271afb85d7ecc33272e0ada2e..83b375bd3c5b5dbd5189d48ad560580883ac91f6 100644
--- a/test/parallel/test-v8-stats.js
+++ b/test/parallel/test-v8-stats.js
@@ -42,11 +42,12 @@ const expectedHeapSpaces = [
   'code_large_object_space',
   'code_space',
   'large_object_space',
-  'map_space',
   'new_large_object_space',
   'new_space',
   'old_space',
   'read_only_space',
+  'shared_large_object_space',
+  'shared_space'
 ];
 const heapSpaceStatistics = v8.getHeapSpaceStatistics();
 const actualHeapSpaceNames = heapSpaceStatistics.map((s) => s.space_name);
